src/game.h

```diff
@@ -486,8 +488,12 @@ class game
         void reload_npcs();
         /** Returns the number of kills of the given mon_id by the player. */
         int kill_count( const mtype_id &id );
-        /** Increments the number of kills of the given mtype_id by the player upwards. */
-        void increase_kill_count( const mtype_id &id );
+        /** Returns the number of kills of the given monster species by the player. */
+        int species_kill_count( const species_id &spec );
+        /** Increments the number of kills of the given mtype_id by the player upwards.
+         * Species defaults to ZOMBIE.
+         */
+        void increase_kill_count( const mtype_id &id, const species_id &spec = species_id( "ZOMBIE" ) );
```

Comment:
This won't work as you think it will. All (existing) calls to this function (and you have not added any new calls to it) only supply one argument, so the `spec` parameter will *always* be "ZOMBIE", regardless of the monster type. Anyway: summing up the kills from the existing map means you won't have to change this.

Reply:
> All (existing) calls to this function (and you have not added any new calls to it) only supply one argument, so the spec parameter will always be "ZOMBIE", regardless of the monster type.

For now there's no need for determining the species in one existing call of this function (in `monster::die`), so it defaults to `ZOMBIE` for the need of `Kill 100 Zombies` quest. If we ever need to determine the species for some other thing, we can use the `spec` parameter, something like 

```
if( monster->in_species( BLOB ) ) {
    g->increase_kill_count( type->id, BLOB );
}
```

Reply:
>For now there's no need for determining the species in one existing call of this function (in monster::die), so it defaults to ZOMBIE for the need of Kill 100 Zombies quest.

> If we ever need to determine the species for some other thing, we can use the spec parameter, something like

How do you know when this happens? The species type of the mission is now loadable from JSON. It can be anything.

So if somebody adds a mission to kill some "BLOB" monsters, it won't work because the only species that is currently counted is "ZOMBIE". Why make it loadable from JSON anyway? The only useful value for that setting is "ZOMBIE".

And to count any other species "X", I have to add another call to this function with "X" as second parameter? That would actually count the monster type again.

Reply:
Okay, reverted monster species loading from JSON.

Reply:
So now the attempt to make this function moddable has been reverted?

> Why make it loadable from JSON anyway? The only useful value for that setting is "ZOMBIE".

That was an attempt to allow other missions in the the future to accept other missions purely from JSON, without having to add more hardcoding. Night didn't fully change the hardcoding to allow for that.

Reply:
> So now the attempt to make this function moddable has been reverted?

I think that I need to make this thing work right first. Making it moddable seems to be fairly easy, so it can wait for another PR.

Reply:
Might be best then.